### DISCUSSION TITLE
[aws-lambda] event object doesn't match AppSyncResolverEvent

### DISCUSSION THREAD

**Discussion by jdanielsyeah** [DefinitelyTyped/DefinitelyTyped#62590] [Issues with a @types package]
(Apologies - I don't know how to write an issue specifically for the specific @types package: `aws-lambda`.)

Essentially, my problem is that when I try to use `AppSyncResolverEvent`, in `import { AppSyncResolverEvent } from "aws-lambda";` I find that the interface is different from the event object I'm receiving in my AWS Lambda. I've provided further details in a stackoverflow question: [How to type the event of a serverless resolver sent from an Amplify GraphQL call](https://stackoverflow.com/questions/73974440/how-to-type-the-event-of-a-serverless-resolver-sent-from-an-amplify-graphql-call)

Again, sorry that I've probably done this incorrectly. I've not added an issue to DefinitelyTyped before.

[a comment by an automated account is not shown]

**Comment by jamesorlakin** [comment]
From what I gather from snippets of documentation I think the behaviour described in the Stack Overflow question is correct. Though I'll be honest and say I have no practical experience with AppSync!

The payload you're receiving to Lambda seems to match that in the [Amplify documentation](https://docs.amplify.aws/cli/graphql/custom-business-logic/#structure-of-the-function-event). In the later 'how it works' section it refers to creating:
> 3. An AWS AppSync pipeline function that prepares the lambda event and invokes the new data source.

Which leads me to think that Amplify is the one defining the mapping that AppSync is passing through to Lambda instead of what AppSync might natively do. There's some documentation within AppSync on ['direct'](https://docs.aws.amazon.com/appsync/latest/devguide/direct-lambda-reference.html) Lambda resolvers which is what an `AppSyncResolverEvent` really refers to, but I guess this is a red herring in your scenario.

I'll have a little think overnight on whether adding this Amplify trigger is right for the `aws-lambda` package. If you have more detailed examples to hand of the payload shape it would be much appreciated.

**Reply by jdanielsyeah** [reply]
Thanks for looking into this James. I'll leave it up to you to decide whether it's appropriate, as I'm brand new to this and wouldn't know. If it's any use, I've hard-coded my event, which takes this shape:
```js

export interface EventJD1 {
  typeName: string;
  fieldName: string;
  arguments: object;
  identity: {
    claims: {
      sub: string;
      "cognito:groups": [string];
      iss: string;
      client_id: string;
      origin_jti: string;
      event_id: string;
      token_use: string;
      scope: string;
      auth_time: number;
      exp: number;
      iat: number;
      jti: string;
      username: string;
    };
    defaultAuthStrategy:  string;
    groups: [string];
    issuer:
     string;
    sourceIp: [string];
    sub:  string;
    username:  string;
  };
  source: null;
  request: {
    headers: {
      "x-forwarded-for": string;
      "sec-ch-ua-mobile": string;
      "cloudfront-viewer-country": string;
      "cloudfront-is-tablet-viewer": string;
      "x-amzn-requestid": string;
      via: string;
      "cloudfront-forwarded-proto": string;
      origin: string;
      "content-length": string;
      "x-forwarded-proto": string;
      "accept-language": string;
      host: string;
      "user-agent": string;
      "cloudfront-is-mobile-viewer": string;
      accept: string;
      "cloudfront-viewer-asn": string;
      "cloudfront-is-smarttv-viewer": string;
      "accept-encoding": string;
      referer: string;
      "content-type": string;
      "sec-fetch-mode": string;
      "x-amzn-trace-id": string;
      "x-amz-cf-id": string;
      authorization: string;
      "sec-fetch-dest": string;
      "x-amz-user-agent": string;
      "sec-ch-ua-platform": string;
      "cloudfront-is-desktop-viewer": string;
      "sec-fetch-site": string;
      "sec-ch-ua": string;
      "x-forwarded-port": string;
    };
    domainName: null;
  };
  prev: { result: any };
}
```

As you said, this is as per the docs you linked to. Thanks again for the attention and response!

**Reply by jamesorlakin** [reply]
I've added this in #62643
